hibernate-cache-local/src/main/java/org/infinispan/tutorial/simple/hibernate/cache/local/InfinispanHibernateCacheLocal.java

```diff
@@ -0,0 +1,266 @@
+      // * a query cache hit
+      stats = queryEntities(emf);
+      System.out.printf("Event entity cache hits: %d (expected 6)%n", stats.getSecondLevelCacheHitCount());
+      System.out.printf("Query cache hit: %d (expected 1)%n", stats.getQueryCacheHitCount());
+
+      // Update one of the persisted entities, stats should show a cache hit and a cache put
+      cacheStats = updateEntity(2L, emf);
+      System.out.printf("Event entity cache hits: %d (expected 7)%n", cacheStats.getHitCount());
+      System.out.printf("Event entity cache puts: %d (expected 6)%n", cacheStats.getPutCount());
+
+      // Repeat query after update, expect:
+      // * no cache hits or puts since entities are already cached
+      // * a query cache miss and query cache put, because when an entity is updated,
+      //   any queries for that type are invalidated
+      stats = queryEntities(emf);
+      System.out.printf("Query cache miss: %d (expected 2)%n", stats.getQueryCacheMissCount());
+      System.out.printf("Query cache put: %d (expected 2)%n", stats.getQueryCachePutCount());
+
+
+      // Save cache-expiring entity, stats should show a second level cache put
+      cacheStats = saveExpiringEntity(emf);
+      System.out.printf("Person entity cache puts: %d (expected 1)%n", cacheStats.getPutCount());
+
+      // Find expiring entity, stats should show a second level cache hit
+      cacheStats = findExpiringEntity(emf);
+      System.out.printf("Person entity cache hits: %d (expected 1)%n", cacheStats.getHitCount());
+
+      // Wait long enough for entity to be expired from cache
+      Thread.sleep(1100);
+
+      // Find expiring entity, after expiration entity should come from DB
+      // Stats should show a cache miss and a cache put
+      cacheStats = findExpiringEntity(emf);
+      System.out.printf("Person entity cache miss: %d (expected 1)%n", cacheStats.getMissCount());
+      System.out.printf("Person entity cache put: %d (expected 2)%n", cacheStats.getPutCount());
+
+      // Close persistence manager
+      emf.close();
+   }
+
+   private static SecondLevelCacheStatistics persistEntities(EntityManagerFactory emf) {
+      EntityManager em = emf.createEntityManager();
+      EntityTransaction tx = em.getTransaction();
+      try {
+         tx.begin();
+
+         em.persist(new Event("Caught a pokemon!"));
+         em.persist(new Event("Hatched an egg"));
+         em.persist(new Event("Became a gym leader"));
+
+         return getCacheStatistics(Event.class.getName(), em);
+      } catch (Throwable t) {
+         tx.setRollbackOnly();
+         throw t;
+      } finally {
+         if (tx.isActive()) tx.commit();
+         else tx.rollback();
+
+         em.close();
+      }
+   }
+
+   private static SecondLevelCacheStatistics findEntity(EntityManagerFactory emf) {
+      EntityManager em = emf.createEntityManager();
+      try {
+         Event event = em.find(Event.class, 1L);
```

Comment:
Relying on auto-generated ids does not seem stable (especially since you need to provide the `L` suffix), but probably good enough for tutorial. I would pass `1L` as a parameter, though, to be symmetric with `updateEntity`.